src/inttest/java/com/faforever/api/login/LoginTest.java

```diff
@@ -0,0 +1,73 @@
+package com.faforever.api.login;
+
+import com.faforever.api.AbstractIntegrationTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class LoginTest extends AbstractIntegrationTest {
+  private MultiValueMap<String, String> params;
+
+  @Override
+  @Before
+  public void setUp() {
+    super.setUp();
+    params = new LinkedMultiValueMap<>();
+    params.add("grant_type", "password");
+    params.add("client_id", "test");
+    params.add("client_secret", "test");
+    params.add("username", "MODERATOR");
+    params.add("password", "MODERATOR");
+  }
+
+  @Test
+  public void retrieveModeratorToken() throws Exception {
+    performOAuthTokenRequest(params)
+      .andExpect(status().isOk())
+      .andExpect(jsonPath("$.access_token", is(notNullValue())));
+  }
+
+  @Test
+  public void wrongUserName() throws Exception {
+    params.set("username", "invalid_password");
```

Comment:
for the sake of consistency make it invalid_username ?

Reply:
copy paste coding... :D